lib/rjgit.rb

```diff
@@ -163,6 +168,28 @@ def self.diff(repository, options = {})
       RJGit.convert_diff_entries(diff_entries)
     end
 
+    def self.describe(repository, ref, options = {})
+      options = {:always => false, :long => false, :tags => false, :match => []}.merge(options)
+      repo = RJGit.repository_type(repository)
+      git = RubyGit.new(repo).jgit
+      command = git.describe.
```

Comment:
Since we return `nil` for all exceptions, we might as well start the `begin` clause here and wrap around all the command-related calls, and have one rescue clause (rescuing IncorrectObjectTypeException, IOException, MissingObjectException, RefNotFoundException _and_ InvalidPatternException and returning `nil`).

Reply:
I actually prefer not wrapping more than the part that can raise, so that the code communicates what method calls might throw what exceptions.  It's your project so I'll let you make the final call; I'm sharing my opinion just so that all the pros and cons are considered.

Reply:
Good point. This also makes sense if we want to do something more meaningful with the raised exceptions in the future.